pkg/analysis/utils/serialization/testdata/src/pointers_always/pointer_to_slice_with_minzero.go

```diff
@@ -0,0 +1,34 @@
+package a
+
+type TestPointerToSliceWithMinZeroAlways struct {
+	// +kubebuilder:validation:MinItems=0
+	PtrArrayWithZeroMinItems *[]string `json:"ptrArrayWithZeroMinItems,omitempty"`
```

Comment:
Can we have a case the same as this one but without the pointer? So that the linter tells us to add the pointer?

Reply:
Alright, so it's a bit more complex now. I summarized the new overall logic in the main comment of this PR.